tests/test_responsetypes.py

```diff
@@ -72,17 +72,35 @@ def test_from_headers(self):
             retcls = responsetypes.from_headers(source)
             assert retcls is cls, f"{source} ==> {retcls} != {cls}"
 
-    def test_from_args(self):
-        # TODO: add more tests that check precedence between the different arguments
+    def test_from_args_pre_xtractmime(self):
         mappings = [
             ({'url': 'http://www.example.com/data.csv'}, TextResponse),
-            # headers takes precedence over url
             ({'headers': Headers({'Content-Type': ['text/html; charset=utf-8']}),
-              'url': 'http://www.example.com/item/'}, HtmlResponse),
+              'url': 'http://www.example.com/item/'}, HtmlResponse), # Failing with xtractmime, returning TextResponse expected HtmlResponse
             ({'headers': Headers({'Content-Disposition': ['attachment; filename="data.xml.gz"']}),
-              'url': 'http://www.example.com/page/'}, Response),
-
+              'url': 'http://www.example.com/page/'}, Response), # Failing with xtractmime, returning TextResponse expected Response
```

Comment:
Will it be better to use the existing methods as-is to determine content-type using Content-Disposition, URL, or filename. Or implement all the logics in a separate function??

Reply:
Since the current logic for this is split into several methods, and from the one parsing headers we only want the Content-Disposition part, I think a new, private method may be easier to read (and slightly more efficient, although I don’t expect that to be a concern here).